synths/default-synths.scd

```diff
@@ -75,3 +75,197 @@ SynthDef(\gabor, { |out, sustain = 0.03, note = 0, amp = 1, pan = 0, width = 0.3
+		13.710, 13.715, 15.057, 15.484, 16.469, 16.817, 18.628]**1.0;
+	var tscale = 100.0 / basefreq / (freqlist**(2-clip(decay,0,2)));
+	var ascale =freqlist**clip(voice,0,4);
+	var sound = Mix.arFill(15, {arg i; EnvGen.ar(Env.perc(0.01*tscale[i], 0.5*tscale[i], 0.2*ascale[i] ), timeScale:dur*5)
+		* SinOsc.ar(basefreq * freqlist[i] * Line.kr(1, 1+accelerate))});
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan))
+}).add;
+
+// hooking into a nice synth piano already in supercollider
+// uses the "velocity" parameter to affect how hard the keys are pressed
+// "speed" controls envelope and decay time
+SynthDef(\superpiano,{|out, speed=1, begin=0, end=1, pan, velocity=1, detune=0.1, muffle=1, stereo=0.2, n |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.linen(0.002, dur, 0.002, 1,-3), timeScale:dur, doneAction:2);
+	// the +0.01 to freq is because of edge case rounding internal to the MdaPiano synth
+	var sound = MdaPiano.ar(n.midicps+0.01, vel:velocity*100, hard:0.8*velocity, decay:0.4*dur,
+		tune:0.5, random:0.05, stretch:detune, muffle:0.8*muffle, stereo:stereo);
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env))
+}).add;
+
+// waveguide mesh, hexagonal drum-like membrane
+SynthDef(\superhex,{|out, speed=1, begin=0, end=1, pan, n, accelerate |
+	var dur = (end-begin)/speed;
+	var env = EnvGen.ar(Env.linen(0.02, dur, 0.02, 1,-3), timeScale:dur*speed, doneAction:2);
+	var tension = 0.05*n.midicps/400 * Line.kr(1,accelerate+1);
+	var loss = 1.0 - (0.01 * speed / n.midicps);
+	var sound = MembraneHexagon.ar(Decay.ar(Impulse.ar(0,0,1), 0.01), tension, loss);
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env))
+}).add;
+
+// Kick Drum using Rumble-San's implementation as a starting point
+// http://blog.rumblesan.com/post/53271713518/drum-sounds-in-supercollider-part-1
+// "n" controls the kick frequency
+// "speed" affects overall envelope timescale, "accelerate" sweeps the click filter freq,
+// "pitch1" affects the click frequency, and "decay" changes the click duration relative to the overall timescale
+SynthDef(\superkick, {|out, speed=1, begin=0, end=1, pan, accelerate, n, pitch1=1, decay=1 |
+	var env, sound, dur, clickdur;
+	dur = 1.0 * (end - begin)/speed;
+	env = EnvGen.ar(Env.linen(0.01, 0, 0.5, 1, -3), timeScale:dur, doneAction:2);
+	sound = SinOsc.ar((n+34.5).midicps);
+	clickdur = 0.02*dur*decay;
+	sound = sound + (LPF.ar(WhiteNoise.ar(1), 1500*pitch1*Line.kr(1,1+accelerate,clickdur) ) * Line.ar(1, 0, clickdur));
+	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env))
+}).add;
+
+// A vaguely 808-ish kick drum
+// "n" controls the chirp frequency, "speed" the filter sweep speed, and "voice" the sinewave feedback
+SynthDef(\super808, {|out, speed=1, begin=0, end=1, pan, voice=0, n |
+	var env, sound, dur, freq;
+	dur = 1.0 * (end - begin)/speed;
+	n = ((n>0)*n) + ((n<1)*3);
```

Comment:
Here I'm using `n` more like a sample index.  Originally I just had `n.midicps`, but having all possible available pitches available for the drum seemed overkill, and what sounded best to me was around `30.midicps`.  So I put in an awful hack to have n=0 correspond to 30.midicps so the default `super808` would sound good, and other `n` values skip notes 10 at a time.